roles/confluent.kafka/tasks/main.yml

```diff
@@ -1,139 +1,139 @@
 - name: broker group
   group:
-    name: "{{kafka.broker.group}}"
+    name: "{{kafka.group}}"
 - name: broker user
   user:
-    name: "{{kafka.broker.user}}"
+    name: "{{kafka.user}}"
     comment: "Kafka User"
     system: yes
-    group: "{{kafka.broker.group}}"
+    group: "{{kafka.group}}"
 - name: Change keytabs ownership
   file:
-    path: "{{kafka.broker.keytab}}"
-    group: "{{kafka.broker.group}}"
+    path: "{{kerberos.keytab_dir}}"
+    group: "{{kafka.group}}"
     mode: '650'
     recurse: yes
 - name: create broker data directories
   file:
     path: "{{item}}"
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
     state: directory
     mode: 0755
-  with_items: "{{kafka.broker.datadir}}"
+  with_items: "{{kafka.datadir}}"
 - name: broker plaintext config
   template:
     src: server.properties.j2
-    dest: "{{kafka.broker.config_file}}"
+    dest: "{{kafka.config_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "plaintext"
   notify:
     - restart kafka
 - name: broker ssl config
   template:
     src: server_ssl.properties.j2
-    dest: "{{kafka.broker.config_file}}"
+    dest: "{{kafka.config_file}}"
     mode: 0640
-    owner: "{{kafka.broker.user}}"
-    group: "{{kafka.broker.group}}"
+    owner: "{{kafka.user}}"
+    group: "{{kafka.group}}"
   when: security_mode == "ssl"
   notify:
     - restart kafka
 - name: broker sasl_ssl config
   template:
     src: server_sasl_ssl.properties.j2
-    dest: "{{kafka.broker.config_file}}"
+    dest: "{{kafka.config_file}}"
```

Comment:
Recommend leaving in 'broker' as kafka is not a component but the platform name.